### ISSUE TITLE
 pickled and unpickled h5pickle Dataset cannot be re-pickled

### ISSUE THREAD

**Issue by nup002** [Exteris/h5pickle#10]
Hello Exteris,
First of all, thanks for your work. The h5pickle module is very useful to me.
Now onto my problem:
Pickling a h5pickle Dataset that has been previously pickled and unpickled fails.

Steps to reproduce:
```
import h5pickle as h5py
import pickle
file = h5py.File("my_hdf5_file.hdf5",'r')
dataset = file['Path_to_dataset']
dataset_pickled = pickle.dumps(dataset)
dataset_unpickled = pickle.loads(dataset_pickled)
dataset_repickled = pickle.dumps(dataset_unpickled)
```

The following error is reported:
```
Traceback (most recent call last):

  File "<ipython-input-9-efad5fcf0147>", line 1, in <module>
    dset_repickled = pickle.dumps(dset_unpickled)

  File "C:\Users\magne.lauritzen\AppData\Local\Continuum\anaconda3\lib\site-packages\h5pickle\__init__.py", line 65, in __getstate__
    return {'name': self.name, 'file': self.file_info}

AttributeError: 'Dataset' object has no attribute 'file_info'
```

Are you aware of this problem and can you suggest a workaround?

**Comment by DaanVanVugt** [comment]
Hi, I think I forgot to set the file handle on unpickled groups and datasets
This has been fixed in #11 which I'll create a new release for today.
Note that this requires at least pickle protocol 2, but the highest protocol is recommended.

**Comment by nup002** [comment]
Excellent, good to hear you're on it.
I will be using this module extensively and will report any errors I encounter.

**Comment by DaanVanVugt** [comment]
I'm glad to hear it is useful for you. Any issues, contributions or suggestions are more than welcome.
I have just released 0.4.2 which has these changes applied.
